File: onion-architecture/java/com/example/domain/Student.java
package com.example.domain;

import java.util.HashSet;
import java.util.Set;

public class Student {
    private Long id;
    private String name;
    private Set<Course> enrolledCourses = new HashSet<>();

    public Student(Long id, String name) {
        this.id = id;
        this.name = name;
    }

    public void enroll(Course course) {
        if (enrolledCourses.contains(course)) {
            throw new IllegalStateException("Already enrolled in course");
        }
        enrolledCourses.add(course);
    }

    public Set<Course> getEnrolledCourses() {
        return enrolledCourses;
    }

    public Long getId() {
        return id;
    }
}

File: onion-architecture/kotlin/src/main/kotlin/com/example/domain/Student.kt
package com.example.domain

class Student(
    val id: Long,
    private val name: String
) {
    private val enrolledCourses: MutableSet<Course> = mutableSetOf()

    fun enrol(course: Course) {
        if (enrolledCourses.contains(course)) {
            throw IllegalStateException("Already enrolled in course")
        }
        enrolledCourses.add(course)
    }

    fun getEnrolledCourses(): Set<Course> = enrolledCourses.toSet()
}